### PR TITLE
Improved element drag and drop UI

### PR DESCRIPTION
**Pull request by wassgha**
Partial for #95

### Changes
- Adds snap guidelines (bounds and center of the screen as well as neighbor element guidelines)
Screen Shot 2020-01-30 at 11 22 56 PM

- Improves UI for the drag and drop handles (follows UI mocks)
Screen Shot 2020-01-30 at 11 17 14 PM

- Improves UI for the cropping tool (follows UI mocks)
Screen Shot 2020-01-30 at 11 17 38 PM
- Prevents controls from being shown while dragging

### CODE REVIEW

**Comment by swissspidy** [comment]
> Improves UI for the cropping tool (follows UI mocks)

Looks like the styling for the range slider there and the one in the grid view is identical. We should extract those to a common component that is shared between the two. (can be in a separate PR of course)

**Comment by dvoytenko** [comment]
> Looks like the styling for the range slider there and the one in the grid view is identical. We should extract those to a common component that is shared between the two. (can be in a separate PR of course)

Agree that should be a separate PR since we're focusing on Moveable here.

File: assets/src/edit-story/components/canvas/singleSelectionMovable.js [hunk 1]
```diff
@@ -18,6 +18,7 @@
  * External dependencies
  */
 import PropTypes from 'prop-types';
+import { omit, values } from 'lodash';
```
Comment (dvoytenko) [inline assets/src/edit-story/components/canvas/singleSelectionMovable.js]: I think I heard there was a decision not to use `lodash`. Elsewhere I'm seeing `objectWithout` utils and such.

/cc @barklund

File: assets/src/edit-story/components/canvas/singleSelectionMovable.js [hunk 2]
```diff
@@ -40,6 +41,7 @@ const ALL_HANDLES = [ 'n', 's', 'e', 'w', 'nw', 'ne', 'sw', 'se' ];
 function SingleSelectionMovable( {
 	selectedElement,
 	targetEl,
+	nodesById,
```
Comment (dvoytenko) [inline assets/src/edit-story/components/canvas/singleSelectionMovable.js]: We are directly accessing `useCanvas()` here anyway. So I'd rather get `nodesById` from there than passing it around. Could you also do the same for `MultiSelectionMovable`?

File: assets/src/edit-story/components/canvas/singleSelectionMovable.js [hunk 3]
```diff
@@ -54,6 +56,11 @@ function SingleSelectionMovable( {
 		maxFontSize: dataToEditorY( MAX_FONT_SIZE ),
 	};
 
+	const {
```
Comment (dvoytenko) [inline assets/src/edit-story/components/canvas/singleSelectionMovable.js]: Merge with `useCanvas()` above.

File: assets/src/edit-story/components/canvas/singleSelectionMovable.js [hunk 4]
```diff
@@ -218,13 +226,26 @@ function SingleSelectionMovable( {
 			pinchable={ true }
 			keepRatio={ 'image' === selectedElement.type && isResizingFromCorner }
 			renderDirections={ ALL_HANDLES }
+			snappable={ true }
```
Comment (dvoytenko) [inline assets/src/edit-story/components/canvas/singleSelectionMovable.js:228]: Are you planning to do the same with `MultiSelectionMovable` later? If so, please rename the PR to reflect this.

File: assets/src/edit-story/components/canvas/singleSelectionMovable.js [hunk 5]
```diff
@@ -218,13 +226,26 @@ function SingleSelectionMovable( {
...
+			snapElement={ true }
+			snapHorizontal={ true }
+			snapVertical={ true }
+			snapCenter={ true }
+			horizontalGuidelines={ box.rotationAngle === 0 ?
+				[ 0, canvasHeight / 2, canvasHeight ] :
+				[ canvasHeight / 2 ] }
```
Comment (dvoytenko) [inline assets/src/edit-story/components/canvas/singleSelectionMovable.js]: I thought we saw that rotated shape and guidelines were working fairly well. No?

File: assets/src/edit-story/components/movable/moveStyle.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+	.default-movable .moveable-control,
+	.default-movable .moveable-line.moveable-rotation-line .moveable-control {
+		background: #1a73e8 !important;
+		border: 2px solid #fff !important;
+		margin-left: -6px;
+		box-shadow: 0 2px 5px rgba(0, 0, 0, 0.15);
+	}
+
+	.default-movable .moveable-direction.moveable-n {
+		display: none !important;
```
Comment (dvoytenko) [inline assets/src/edit-story/components/movable/moveStyle.js:32]: Why is this?

File: assets/src/edit-story/elements/image/scalePanel.js [hunk 7]
```diff
@@ -46,8 +46,8 @@ const Container = styled.div`
 	width: ${ ( { width } ) => `${ Math.max( width, MIN_WIDTH ) }px` };
 	height: ${ HEIGHT }px;
 
-	background: ${ ( { theme } ) => theme.colors.bg.v7 };
-	border-radius: 4px;
+	background: ${ ( { theme } ) => theme.colors.bg.v0 + 'EE' };
```
Comment (dvoytenko) [inline assets/src/edit-story/elements/image/scalePanel.js]: Hmm. I'm not a big fun of this. What if the color is defined via `rgba()`? Or already has transparency? Let's just bring it in the theme as a separate constant for now.

File: assets/src/edit-story/elements/image/scalePanel.js [hunk 8]
```diff
@@ -46,8 +46,8 @@ const Container = styled.div`
...
+	border-radius: 100px;
```
Comment (dvoytenko) [inline assets/src/edit-story/elements/image/scalePanel.js:50]: Does this look ok with a shorter-width panel?

File: assets/src/edit-story/elements/image/scalePanel.js [hunk 9]
```diff
@@ -66,18 +66,40 @@ const Range = styled.input.attrs( {
 	margin: 4px;
 	min-width: 100px;
 	cursor: pointer;
+	outline: none;
+	background: #fffc;
```
Comment (dvoytenko) [inline assets/src/edit-story/elements/image/scalePanel.js]: theme

**Review by dvoytenko — CHANGES_REQUESTED** [review]

File: assets/src/edit-story/components/canvas/singleSelectionMovable.js (hunk 1 above)
Comment (swissspidy) [reply assets/src/edit-story/components/canvas/singleSelectionMovable.js]: See https://xwp.slack.com/archives/CCCAUAH6F/p1576519153039000 for that discussion.

File: assets/src/edit-story/components/movable/moveStyle.js (hunk 6 above)
Comment (wassgha) [reply assets/src/edit-story/components/movable/moveStyle.js:32]: This is coming from the UI mocks, the north anchor is not shown

File: assets/src/edit-story/elements/image/scalePanel.js (hunk 8 above)
Comment (wassgha) [reply assets/src/edit-story/elements/image/scalePanel.js:50]: Yup

File: assets/src/edit-story/components/canvas/singleSelectionMovable.js (hunk 4 above)
Comment (wassgha) [reply assets/src/edit-story/components/canvas/singleSelectionMovable.js:228]: Done

File: assets/src/edit-story/app/index.js [hunk 10]
```diff
@@ -44,6 +45,7 @@ function App( { config } ) {
 							<FontProvider>
 								<GlobalStyle />
 								<CropMoveableGlobalStyle />
+                <DefaultMoveableGlobalStyle />
```
Comment (swissspidy) [inline assets/src/edit-story/app/index.js]: tabs

File: assets/src/edit-story/components/movable/moveStyle.js (hunk 6 above)
Comment (dvoytenko) [reply assets/src/edit-story/components/movable/moveStyle.js:32]: Got it.

File: assets/src/edit-story/components/canvas/multiSelectionMovable.js [hunk 11]
```diff
@@ -146,13 +150,17 @@ function MultiSelectionMovable( { selectedElements, nodesById } ) {
...
 		<Movable
+			className="default-movable"
 			ref={ moveable }
 			zIndex={ 0 }
 			target={ targetList.map( ( { node } ) => node ) }
 
 			draggable={ true }
-			resizable={ true }
-			rotatable={ true }
+			// Making resizable depend on state caused a bug where the
```
Comment (dvoytenko) [inline assets/src/edit-story/components/canvas/multiSelectionMovable.js:159]: Link the bug filed at the source?

**Review by dvoytenko — APPROVED** [review]
LGTM. Just one request.

**Pull request merged**